lib/friends_badge/views/friends_badge_page.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_and_friends/friends_badge/friends_badge.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:friends_badge/friends_badge.dart';
+
+class FriendsBadgePage extends StatelessWidget {
+  const FriendsBadgePage({super.key});
+
+  static Route<void> route() {
+    return MaterialPageRoute(builder: (_) => const FriendsBadgePage());
+  }
+
+  @override
+  Widget build(BuildContext context) => const FriendsBadgeView();
+}
+
+class FriendsBadgeView extends StatelessWidget {
+  const FriendsBadgeView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
```

Comment:
The BlocProvider should be moved into the parent (the page) and then you don’t need a Builder below